astropy/time/tests/test_delta.py

```diff
@@ -297,6 +298,45 @@ def test_scales_for_time_minus_time(self, scale1, scale2):
         t2_recover = getattr(t2_recover_t1_scale, scale2)
         assert allclose_jd(t2_recover.jd, t2.jd)
 
+    def test_local_scales_for_time_minus_time(self):
+        """T(X) - T2(Y)  -- does T(X) - T2(Y).X and return dT(X)
+        and T(X) +/- dT(Y)  -- does (in essence) (T(X).Y +/- dT(Y)).X
+
+        I.e., time differences of two times should have the scale of the
+        first time.
+
+        Also tests that subtracting two time scales, one of which is
+        different from local time scale should raise TypeError.
+        """
+        scale1 = 'local'
+        scale2 = 'local'
+        t1 = self.t[scale1]
+        t2 = Time('2010-01-01', scale=scale2)
+        dt = t1 - t2
+        assert dt.scale == scale1
+
+        # now check with delta time; also check reversibility
+        t1_recover_t2_scale = t2 + dt
+        assert t1_recover_t2_scale.scale == scale2
+        t1_recover = getattr(t1_recover_t2_scale, scale1)
```

Comment:
This is a pointless operation when the scale is already 'local'; just call the above `t1_recover` instead of `t1_recover_t2_scale`.